pkg/v1/tkg/tkgctl/helpers.go

```diff
@@ -185,37 +190,141 @@ func (t *tkgctl) checkIfInputFileIsClusterClassBased(clusterConfigFile string) (
-					for k := range variables[j].(map[string]interface{}) {
-						if k == "name" {
-							name = variables[j].(map[string]interface{})[k].(string)
-						} else {
-							value = fmt.Sprintf("%v", variables[j].(map[string]interface{})[k])
-						}
-					}
-					if len(value) > 0 {
-						variablesMap[name] = value
-					}
+func (t *tkgctl) processCClusterObjectForConfigurationVariables(cclusterObj unstructured.Unstructured) error {
+	inputVariablesMap := make(map[string]interface{})
+	inputVariablesMap["metadata.name"] = cclusterObj.GetName()
+	inputVariablesMap["metadata.namespace"] = cclusterObj.GetNamespace()
+	spec := cclusterObj.Object[constants.SPEC].(map[string]interface{})
+	err := processYamlObjectAndAddToMap(spec, constants.SPEC, inputVariablesMap)
+	if err != nil {
+		return err
+	}
+	providerName, err := validateTopologyClassName(inputVariablesMap[constants.TopologyClass].(string))
+	if err != nil {
+		return err
+	}
+	legacyVarMap := make(map[string]string)
+	ccToLegacyNamesMap := constants.InfrastructureSpecificVariableMappingMap[providerName]
+	// assign cluster class input values to legacy variables
+	for inputVariable := range inputVariablesMap {
+		if legacyNameForClusterObjectInputVariable, ok := ccToLegacyNamesMap[inputVariable]; ok {
+			legacyVarMap[legacyNameForClusterObjectInputVariable] = fmt.Sprintf("%v", inputVariablesMap[inputVariable])
+		}
+	}
+	// override legacy variables with higher precedence values if exists
+	if providerName == constants.InfrastructureProviderAWS {
+		for higherPrecedenceKey := range constants.CclusterAttributesHigherPrecedenceToLowerMap {
+			legacyName, ok1 := ccToLegacyNamesMap[higherPrecedenceKey]
+			value, ok2 := inputVariablesMap[higherPrecedenceKey]
+			if ok1 && ok2 {
+				legacyVarMap[legacyName] = fmt.Sprintf("%v", value)
+			}
+		}
+	}
+
+	t.TKGConfigReaderWriter().SetMap(legacyVarMap)
+	return nil
+}
+
+/*
+ processYamlObjectAndAddToMap process specific value of the Cluster yaml object
+ value is the value for the given attribute path ccMappingName in Cluster object
+ if input value is final child value then value is added to variablesMap with ccMappingName as key,
+ if the input value is not final child value, then its process again for next level child.
+*/
+func processYamlObjectAndAddToMap(value interface{}, ccMappingName string, variablesMap map[string]interface{}) error {
+	var err error
+	switch value := value.(type) {
+	case []interface{}:
+		err = processYamlObjectArrayInterfaceType(value, ccMappingName, variablesMap)
+	case []map[string]interface{}:
+		for index := range value {
+			err = processYamlObjectAndAddToMap(value[index], ccMappingName, variablesMap)
+			if err != nil {
+				return err
+			}
+		}
+	case map[string]interface{}:
+		for key := range value {
+			nextLevelName := key
+			nextLevelVal := value[nextLevelName]
+			// noProxy has value of type array, so assign it here
+			if ccMappingName+"."+nextLevelName == "spec.topology.variables.proxy.noProxy" {
```

Comment:
In Cluster object, the "spec.topology.variables.proxy.noProxy" has value type as Array, which needs to be assigned as it is, otherwise function "processYamlObjectAndAddToMap" process each index value in array as separate value and overrides for the key "spec.topology.variables.proxy.noProxy".
eg: 
        noProxy:
        - .svc.cluster.local
        - 169.254.0.0/16
If I don't have the above special case, the value for the "spec.topology.variables.proxy.noProxy" comes as "169.254.0.0/16", which is not correct we need the entire array as a value. that's why I have added that special case!